src/Connect.js

```diff
@@ -1,68 +1,201 @@
+    this.onloadResponse = transport.url.getResponse()
+    transport.url.listenResponse((err, res) => {
+      if (res) this.PubSub.publish(res.id, {res: res.res, data: res.data}) // TODO pass errors
+    })
+    // TODO need a nicer way configure resolvers, lines below should be temporary
+    // TODO would be nice to have some uncoupled funct for parsing/verifying in uport-core-js rather than require config a credential object, nice to require aud either here
+    UportDIDResolver(UportLite({ networks: network.config.networkToNetworkSet(this.network) }))
+    EthrDIDResolver(opts.ethrConfig || {})
+    this.verifyResponse = (res) => {
+      const decodedToken = decodeJWT(res).payload
+      return verifyJWT(res, {callbackUrl: decodedToken.aud}).then(this.credentials.processDisclosurePayload)
+    }
+  }
+
+ /**
+  *  Instantiates and returns a web3 styple provider wrapped with uPort functionality.
+  *  For more details see uportSubprovider. uPort overrides eth_coinbase and eth_accounts
+  *  to start a get address flow or to return an already received address. It also
+  *  overrides eth_sendTransaction to start the send transaction flow to pass the
+  *  transaction to the uPort app.
+  *
+  *  @return          {UportSubprovider}    A web3 style provider wrapped with uPort functionality
+  */
+  getProvider () {
+  const subProvider = new provider({
+    requestAddress: this.requestAddress.bind(this),
+    sendTransaction: this.sendTransaction.bind(this),
+    provider: this.provider,
+    networkId: this.network.id
+  })
+  if (this.address) subProvider.setAccount(this.address)
+  return subProvider
+  }
+
+ /**
+  *  Creates a request for only the address/id of the uPort identity.
+  *
+  *  @param    {String}    [id='addressReq']    string to identify request, later used to get response
+  */
+  requestAddress (id='addressReq') {
+   this.request(simpleRequest(), id)
   }
 
+ // TODO offer listener and single resolve? or other both for this funct, by allowing optional cb instead
  /**
-  *  Instantiates and returns a web3 object wrapped with uPort functionality. For
-  *  more details see uportSubprovider and getProvider in connectCore.
+  *  Get response by id of earlier request, returns promise which resolves when first reponse with given id is avaialable. Listen instead, if looking for multiple responses of same id.
   *
-  *  @return          {web3}    A uPort web3 object
+  *  @param    {String}    id             id of request you are looking for a response for
+  *  @return   {Promise<Object, Error>}   promise resolves once valid response for given id is avaiable, otherwise rejects with error
   */
-  getWeb3 () {
-    const provider = this.getProvider()
-    const web3 = new Web3()
-    web3.setProvider(provider)
-    // Work around to issue with web3 requiring a from parameter. This isn't actually used.
-    web3.eth.defaultAccount = '0xB42E70a3c6dd57003f4bFe7B06E370d21CDA8087'
-    return web3
+  onResponse(id) {
```

Comment:
I think I understand this and it looks really cool. Looking forward to a full explanation.